src/server/modules/inventory.ts

```diff
@@ -72,19 +72,31 @@ export class SellItemEvent extends ServerSocketEvent implements ServerEvent {
 export class SalvageItemEvent extends ServerSocketEvent implements ServerEvent {
   event = ServerEventName.ItemSalvage;
   description = 'Salvage an item in your inventory.';
-  args = 'itemId';
+  args = 'itemId?, itemSlot?';
 
-  async callback({ itemId } = { itemId: '' }) {
+  async callback({ itemId, itemSlot } = { itemId: '', itemSlot: '' }) {
     const player = this.player;
     if(!player) return this.notConnected();
 
-    const foundItem = player.$inventory.getItemFromInventory(itemId);
-    if(!foundItem) return this.gameError('Could not find that item in your inventory.');
+    const foundItem = (function() {
+      if(itemSlot == null) {
```

Comment:
don't check against `null` like this. similar to below.

Reply:
Same as the other null check. Fixed.